src/gc/gc.cpp

```diff
@@ -26685,7 +26683,7 @@ void gc_heap::background_grow_c_mark_list()
         assert (new_c_mark_list);
         memcpy (new_c_mark_list, c_mark_list, c_mark_list_length*sizeof(uint8_t*));
         c_mark_list_length = c_mark_list_length*2;
-        delete c_mark_list;
+        delete [] c_mark_list;
```

Comment:
@jkotas I'm guessing the compilers were not as advanced years ago 'cause Patrick specifically did this way. it's good they are doing it now. 

@sdmaclea reverting this change and creating an issue now sounds great to me. thanks!